include/deal.II/base/tensor_function_parser.h

```diff
@@ -0,0 +1,361 @@
+                 const std::string &variable_names = default_variable_names() +
+                                                     ",t");
+
+  /**
+   * Destructor. Explicitly delete the TensorFunctionParser objects (there is one
+   * for each component of the function).
+   */
+  ~TensorFunctionParser() override;
+
+  /**
+   * Type for the constant map. Used by the initialize() method.
+   */
+  using ConstMap = std::map<std::string, double>;
+
+  /**
+   * Iterator for the constants map. Used by the initialize() method.
+   */
+  using ConstMapIterator = ConstMap::iterator;
+
+  /**
+   * Initialize the tensor function.  This methods accepts the following parameters:
+   *
+   * <b>vars</b>: a string with the variables that will be used by the
+   * expressions to be evaluated. Note that the variables can have any name
+   * (of course different from the function names defined above!), but the
+   * order IS important. The first variable will correspond to the first
+   * component of the point in which the function is evaluated, the second
+   * variable to the second component and so forth. If this function is also
+   * time dependent, then it is necessary to specify it by setting the
+   * <code>time_dependent</code> parameter to true.  An exception is thrown if
+   * the number of variables specified here is different from dim (if this
+   * function is not time-dependent) or from dim+1 (if it is time- dependent).
+   *
+   * <b>expressions</b>: a list of strings containing the expressions that
+   * will be byte compiled by the internal parser (TensorFunctionParser). Note that
+   * the size of this vector must match exactly the number of components of
+   * the TensorFunctionParser, as declared in the constructor. If this is not the
+   * case, an exception is thrown.
+   *
+   * <b>constants</b>: a map of constants used to pass any necessary constant
+   * that we want to specify in our expressions (in the example above the
+   * number pi). An expression is valid if and only if it contains only
+   * defined variables and defined constants (other than the functions
+   * specified above). If a constant is given whose name is not valid (eg:
+   * <code>constants["sin"] = 1.5;</code>) an exception is thrown.
+   *
+   * <b>time_dependent</b>. If this is a time dependent function, then the
+   * last variable declared in <b>vars</b> is assumed to be the time variable,
+   * and this->get_time() is used to initialize it when evaluating the
+   * function. Naturally the number of variables parsed by the initialize()
+   * method in this case is dim+1. The value of this parameter defaults to
+   * false, i.e. do not consider time.
+   */
+  void
+  initialize(const std::string &             vars,
+             const std::vector<std::string> &expressions,
+             const ConstMap &                constants,
+             const bool                      time_dependent = false);
```

Comment:
It seems odd that `t` is automatically added as a variable for the constructor but here we set `time_dependent = false`. We should set these consistently.

Reply:
Nevermind - `FunctionParser` also has this inconsistency. We should match that classes' behavior.